### PR TITLE
Introduce the change detection optimization article

### LINKED ISSUE #1068: content: Optimize Angular's change detection
**A one to two sentence description of your post**
What are you planning to teach folks in this post?

**Target publish date:** `<2019-07-16>`
- [ ] Check this box if this is a hard deadline.

**Process**
Make a copy of [this template](https://docs.google.com/document/d/1ydauCufwwavStaKxhIDHuNivVxgQBOdDno4uLjGIjmE/edit)
and fill in the proposal and outline sections. When you're finished, drop a link
to the doc in the 1st step below and check the box. A member of the web.dev team
will assign this ticket to themselves and work as your reviewer to help you
complete the remaining steps.

- [x] 1. I've done my proposal https://docs.google.com/document/d/1zZ0Yghlkl1vFpXYnPNFdIvImH4YhgTI4Wr3PsxZQ6xg/edit#heading=h.qgw78qcln70
- [x] 2. Proposal approved — _reviewer checks this_
- [x] 3. I've added a draft to my proposal doc
- [x] 4. Draft approved — _reviewer checks this_
- [x] 5. I've submitted a pull request #1134
- [x] 6. Pull request approved — _reviewer checks this_

### PR DESCRIPTION
**Pull request by mgechev**
Fixes #1068.

Content from [here](https://docs.google.com/document/d/1zZ0Yghlkl1vFpXYnPNFdIvImH4YhgTI4Wr3PsxZQ6xg/edit), image from [here](https://pixabay.com/photos/red-background-wallpaper-lighting-2312548/).

### CODE REVIEW

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
+---
+layout: post
+title: "Optimize Angular's change detection"
+subhead: |
+  Faster change detection for better user experience.
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
  Implement faster change detection for better user experience.
```

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+hero: change-detection.jpg
+alt: Different light bulb
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
alt: An array of light bulbs with a single bulb turned on.
```

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+date: 2019-07-09
+description: |
+  Optimize the change detection of your app to make it more responsive and faster to use.
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
  Learn how to optimize your Angular app's change detection to make it more responsive.
```
"faster to use" seems more or less the same as "more responsive"?

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+
+## Inside Angular's change detection
+
+To understand how Angular's change detection works, let's look at a sample app!
+
+_You can find the code for the app in [this GitHub repository](https://github.com/mgechev/change-detection-web-dev)._
+
+
+![Sample application](app.png "Sample application")
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
```

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+
+The app lists employees from two departments in a company—sales and R&D—and has two components:
+
+* `AppComponent`, which is the root component of the app, and
+* Two instances of `EmployeeListComponent`, one for sales and one for R&D.
+
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion

![Sample application](app.png "Sample application")

```
Sorry I didn't catch this before: we usually put the explanation _before_ the image so readers know what they're looking at.

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    src="https://stackblitz.com/github/mgechev/change-detection-web-dev?embed=1&file=src/app/app.component.ts&view=preview"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+You can verify that the slowdown comes from the `fibonacci` function by setting up the [example project](https://github.com/mgechev/change-detection-web-dev) and opening the **Performance** tab of Chrome DevTools.
+
+{% Instruction 'devtools-performance', 'ol' %}
+
+Now click <span style="width: 13px;height: 13px;background-color: #6E6E6E;display: inline-block;border-radius: 50%;margin-left: 3px;margin-right: 3px;"></span> **Record** (in the top-left corner of the **Performance** panel) and start typing in one of the text boxes in the app. In a few seconds, click **Record** [icon] again to stop recording. Once Chrome DevTools processes all the profiling data it collected, you'll see something like this:
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
Now click **Record** <span style="width: 13px;height: 13px;background-color: #6E6E6E;display: inline-block;border-radius: 50%;margin-left: 3px;margin-right: 3px;"></span> (in the top-left corner of the **Performance** panel) and start typing in one of the text boxes in the app. In a few seconds, click **Record** <span style="width: 13px;height: 13px;background-color: #6E6E6E;display: inline-block;border-radius: 50%;margin-left: 3px;margin-right: 3px;"></span> again to stop recording. Once Chrome DevTools processes all the profiling data it collected, you'll see something like this:
```
Icons go after button label names.

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+![Performance profiling](profiling.png "Performance profiling in Chrome DevTools")
+
+If there are many employees in the list, this process may block the browser's UI thread and causes frame drops, which leads to a bad user experience.
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
If there are many employees in the list, this process may block the browser's UI thread and cause frame drops, which leads to a bad user experience.
```

File: src/site/content/en/angular/faster-change-detection/index.md [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+
+## Skipping component subtrees
+
+When the user is typing in the text input for the sales `EmployeeListComponent` you know that the data in the R&D department isn't changing—so there's no reason to run change detection on its component. To make sure the R&D instance doesn't trigger change detection, set the `changeDetectionStrategy` of `EmployeeListComponent` to `OnPush`:
```
Comment (mfriesenhahn) [inline src/site/content/en/angular/faster-change-detection/index.md]:
```suggestion
When the user is typing in the text input for the _sales_ `EmployeeListComponent` you know that the data in the _R&D_ department isn't changing—so there's no reason to run change detection on its component. To make sure the R&D instance doesn't trigger change detection, set the `changeDetectionStrategy` of `EmployeeListComponent` to `OnPush`:
```
I think the extra emphasis makes it easier to follow the logic of the sentence.

**Comment by mgechev** [comment]
@mfriesenhahn it's all done! Thanks for the review!

**Pull request merged**